### PR TITLE
Fix CaffeFunctionDenseNet121

### PR DESCRIPTION
**Pull request by mitmul**
Caffe's BatchNorm has no scaling parameters and it always comes with Scale layer somehow. So, if you convert a Caffe's BatchNorm layer into Chainer's Links using CaffeFunction, it produces two Links; L.BatchNormalization and L.Scale from a single Caffe's BatchNorm. The additional Scale layers consume some more unnecessary memory and spend some time. That slows down the training speed of DenseNet.

This PR just combines those two layers (BN and Scale) into a single BN, and then it improves the computational time in forward pass with ~67%. I haven't checked the backward time but it should be improved too.

(Actually, @okuta found this point. Thanks!)

### CODE REVIEW

**Comment by ilkarman** [comment]
Thanks so much! Will definitely rerun!

**Comment by okuta** [comment]
@mitmul I appreciate your work.

**Comment by ilkarman** [comment]
That has seemed to do the trick! Mem usage is around 10GB similar to PyTorch (so I can use batch 64), speed is a tiny bit faster than PyTorch now. You guys are incredible

**Comment by Crissman** [comment]
Thank you! Can we ask you to update the readme.MD and your Medium blog post to show the updated performance? :)
